### ISSUE TITLE
neither napri-mrcfile-reader nor napri-mrcfile-handler plug-in opens standard mrc files

### ISSUE THREAD

**Issue by anchi2c** [napari/napari#5837]
Both generated similar error as below.  The example mrc file can be found at https://emg.nysbc.org/redmine/projects/leginon/repository/revisions/trunk/entry/leginon/holetemplate.mrc

Sam problem with mrc files from other sources. 2D or 3D.

`File ~/miniconda3/envs/napari/lib/python3.10/site-packages/npe2/io_utils.py:169, in _read(paths=['/Users/anchi.cheng/packages/myami-python3/leginon/holetemplate.mrc'], stack=False, plugin_name='mrcfile-reader', return_reader=True, _pm=<npe2._plugin_manager.PluginManager object>)
    166             return (layer_data, rdr) if return_reader else layer_data
    168 if plugin_name:
--> 169     raise ValueError(
        plugin_name = 'mrcfile-reader'
        paths = ['/Users/anchi.cheng/packages/myami-python3/leginon/holetemplate.mrc']
    170         f"Plugin {plugin_name!r} was selected to open "
    171         + f"{paths!r}, but returned no data."
    172     )
    173 raise ValueError(f"No readers returned data for {paths!r}")

ValueError: Plugin 'mrcfile-reader' was selected to open ['/Users/anchi.cheng/packages/myami-python3/leginon/holetemplate.mrc'], but returned no data.`

System MacBook Pro Chip Apple M2Max
napari: 0.4.17
Platform: macOS-13.3.1-arm64-arm-64bit
System: MacOS 13.3.1
Python: 3.10.11 | packaged by conda-forge | (main, May 10 2023, 19:01:19) [Clang 14.0.6 ]
Qt: 5.15.6
PyQt5: 5.15.7
NumPy: 1.24.3
SciPy: 1.10.1
Dask: 2023.5.0
VisPy: 0.11.0
magicgui: 0.7.2
superqt: unknown
in-n-out: 0.1.7
app-model: 0.1.4
npe2: 0.7.0

plug-in napari-mrcfile-handler  0.0.6
plug-in napari-mrcfile-reader 0.1.3

**Comment by anchi2c** [comment]
mrcfile module from the same conda environment works.

import mrcfile
mrcfile.read('leginon/holetemplate.mrc')
array([[0., 0., 0., ..., 0., 0., 0.],
       [0., 0., 0., ..., 0., 0., 0.],
       [0., 0., 0., ..., 0., 0., 0.],
       ...,
       [0., 0., 0., ..., 0., 0., 0.],
       [0., 0., 0., ..., 0., 0., 0.],
       [0., 0., 0., ..., 0., 0., 0.]], dtype=float32)

**Comment by andy-sweet** [comment]
This probably belongs on https://github.com/alisterburt/napari-mrcfile-reader

@alisterburt : can I trouble you to transfer/move the issue if you think that makes sense?

**Comment by anchi2c** [comment]
Happy to do so. Just to note that both plug-ins failed.  There is a common pattern here.  I will report to both.

**Comment by andy-sweet** [comment]
@anchi2c : another thing potentially worth trying is

```python
import napari_mrcfile_reader
path = 'leginon/holetemplate.mrc'
reader = napari_mrcfile_reader.napari_get_reader(path)
data = reader(path)
```

which might help distinguish between a plugin issue and a napari issue.

**Comment by andy-sweet** [comment]
Sorry for the multiple comments, but I did a bit more digging here, and I think there may be a compatibility problem between napari==0.4.17 and npe2==0.7.0 at least for this case (and maybe others too).

I could not reproduce the problem in my napari==0.4.17 environment initially which had npe==0.6.1. But after installing npe2==0.7.0, I got the same error when both plugins were in my environment and a slightly different one when only napari-mrcfile-reader was installed.

Downgrading to npe2==0.6.2 fixes the issue for me, so that could also work as a temporary workaround.

**Comment by alisterburt** [comment]
Hi both! Thanks for the ping @andy-sweet 🙂 I had noticed some problems in fresh environments on some colleagues' machines but didn't have time to dig into it - I can take a look today 

@andy-sweet given that it seems NPE related let's keep here for now

**Comment by alisterburt** [comment]
Also @anchi2c excited to see you playing with napari! Please do reach out if you'd like input on anything

**Comment by Czaki** [comment]
This issue is a duplication of #5738. @alisterburt you could convert your plugin to npe2 one to solve this issue permanently. 

I do not see or remember any PR to solve this on the `npe2` side so either someone should declare to work on it or we should close it.

**Comment by alisterburt** [comment]
Thanks @Czaki - guess it's time to update that one!

**Comment by DragaDoncila** [comment]
> I do not see or remember any PR to solve this on the npe2 side so either someone should declare to work on it or we should close it.

@Czaki this will not be changed on `npe2` side, as this error is an improvement for `npe2` plugins - previously, the error was not specific about which plugin was tried. This had an unintended side effect due to us relying on the specific error message from `npe2` before trying `npe1` plugins. #5739 should have fixed this problem on `main` - I've checked just then with the file provided by @anchi2c and it opens ok on `main` with `napari-mrcfile-handler`.

@anchi2c @alisterburt , using the `npe2` shim (see screenshot) should fix this without the need to update `napari-mrcfile-handler` - unless there's some other reason the shim is failing here, which we should know about. 

![image](https://github.com/napari/napari/assets/17995243/4f6d27ac-0fb0-4670-901d-7b2ac27480cb)

**Comment by alisterburt** [comment]
cool! Thanks @DragaDoncila

**Comment by Czaki** [comment]
@DragaDoncila We could still examine napari version (if installed) and adjust the message based on this. But maybe it is not worth it.

**Comment by anchi2c** [comment]
npe2==0.6.2 works for me.

**Comment by anchi2c** [comment]
I revert npe2 to 0.7.0  and selected Use npe2 adaptor.  When I try opening the mrc file, it first indicates that "Tried opening with mrcfile-reader, but failed.  It then try napri-mrcfile_handler, but then I get the same "returned no data error" as before.

**Comment by anchi2c** [comment]
Probably both plug-in needs to be updated. @alisterburt please let me know after you have updated.  I will be happy to beta test.

**Comment by DragaDoncila** [comment]
I tried opening `holetemplate.mrc` with napari `main`, using the npe2 adaptor and `npe2==0.7.0` and it works ok for me - `napari-mrcfile-reader` and `napari-mrcfile-handler` both open it. Not sure what's making the difference in your environment @anchi2c 

```
napari: 0.5.0a2.dev237+gad748ba5
Platform: macOS-10.16-x86_64-i386-64bit
System: MacOS 13.3.1
Python: 3.9.16 (main, Mar 8 2023, 04:29:44) [Clang 14.0.6 ]
Qt: 5.15.2
PyQt5: 5.15.9
NumPy: 1.24.3
SciPy: 1.9.1
Dask: 2023.5.0
VisPy: 0.12.2
magicgui: 0.7.2
superqt: unknown
in-n-out: 0.1.7
app-model: 0.1.4
npe2: 0.7.0

OpenGL:
- GL version: 2.1 INTEL-20.5.7
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1440x900, scale 2.0
- screen 2: resolution 3840x2160, scale 1.0

Settings path:
- /Users/ddoncilapop/Library/Application Support/napari/mrc_974d1d78bb8b395f0fc45ae1754a3d79a50deea7/settings.yaml
```

**Comment by psobolewskiPhD** [comment]
I assume they are using 0.4.17, while you're on 0.5.0a2, so assuming #5738 and #5739 are back ported to 0.4.18 then this should be fixed in the next release?

**Comment by alisterburt** [comment]
That's my conclusion too @psobolewskiPhD 

(And I should still update my plugin 😝)

**Comment by alisterburt** [comment]
closed by alisterburt/napari-mrcfile-reader#7
